src/trio/_timeouts.py

```diff
@@ -123,11 +141,14 @@ def fail_at(deadline: float) -> AbstractContextManager[trio.CancelScope]:  # typ
         raise TooSlowError
 
 
-if not TYPE_CHECKING:
-    fail_at = contextmanager(fail_at)
+# This was previously a simple @contextmanager-based cm, but in order to save
+# the current time at initialization it needed to be class-based.
+# Once that change of behaviour has gone through and the old functionality is removed
+# it can be reverted.
```

Comment:
Might be worth putting a comment alongside related deprecation warning because someone might just find usages of the deprecation function and remove them rather than searching more in the file.